indexer/editable_map_object.cpp

```diff
@@ -312,6 +312,13 @@ bool EditableMapObject::ValidateEmail(string const & email)
   if (email.empty())
     return true;
 
+  // Email cannot begins and ends with '.' or '@'
```

Comment:
Why not to check
```
(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)
```
?
You do not need a regexp engine, because you can easily implement the validator by hands, and it will work fine on most cases.